flang/lib/Lower/OpenMP.cpp

```diff
@@ -2213,6 +2212,27 @@ static void createBodyOfTargetDataOp(
     }
     argIndex++;
   }
+
+  // Insert dummy instruction to remember the insertion position. The
+  // marker will be deleted since there are not uses.
+  // In the HLFIR flow there are hlfir.declares inserted above while
+  // setting block arguments.
+  mlir::Value undefMarker = firOpBuilder.create<fir::UndefOp>(
+      dataOp.getOperation()->getLoc(), firOpBuilder.getIndexType());
+
+  // Create blocks for unstructured regions. This has to be done since
+  // the original block allocation are created with the function as
```

Comment:
Nit: either "allocation -> allocations" or "are -> is", not sure which is correct in this context

Reply:
I reworded it to the following.
```
   // Create blocks for unstructured regions. This has to be done since
   // blocks are initially allocated with the function as the parent region.
```

Reply:
Thank you very much @kiranchandramohan